File: openni/c++/ni-simple-read/ni-simple-read.cpp
#include <stdio.h>
#include <openni2/OpenNI.h>

using namespace openni;

int main()
{
  if (OpenNI::initialize() != STATUS_OK)
  {
    printf("Initialize failed\n%s\n", OpenNI::getExtendedError());
    return 1;
  }

  Device device;
  if (device.open(ANY_DEVICE) != STATUS_OK)
  {
    printf("Couldn't open device\n%s\n", OpenNI::getExtendedError());
    return 2;
  }

  VideoStream depth;
  if (depth.create(device, SENSOR_DEPTH) != STATUS_OK)
  {
    printf("Couldn't create depth stream\n%s\n", OpenNI::getExtendedError());
    return 3;
  }

  if (depth.start() != STATUS_OK)
  {
    printf("Couldn't start the depth stream\n%s\n", OpenNI::getExtendedError());
    return 4;
  }

  VideoFrameRef frame;
  while (true)
  {
    if (depth.readFrame(&frame) != STATUS_OK)
    {
      printf("Read failed!\n%s\n", OpenNI::getExtendedError());
      continue;
    }

    DepthPixel *pDepth = (DepthPixel *)frame.getData();

    int middleIndex = (frame.getHeight() + 1) * frame.getWidth() / 2;

    printf("Distance is %5d mm\n", pDepth[middleIndex]);
  }

  depth.stop();
  depth.destroy();
  device.close();
  OpenNI::shutdown();

  return 0;
}

File: openni/python/ni-simple-read.py
from openni import openni2

openni2.initialize()
dev = openni2.Device.open_any()

depth_stream = dev.create_depth_stream()
depth_stream.start()

video_mode = depth_stream.get_video_mode()
middleIndex = int((video_mode.resolutionY + 1) * video_mode.resolutionX / 2)

try:
    while(True):
        frame = depth_stream.read_frame()
        pDepth = frame.get_buffer_as_uint16()
        print(f'''Distance is {pDepth[middleIndex]} mm''')
finally:
    dev.close()
    openni2.unload()
